### PR TITLE
Added Remove-NsxIPPool Function

### PR DESCRIPTION
**Pull request by rhancock8**
Added function to remove an NSX IP Pool

### CODE REVIEW

File: PowerNSX.psm1 [hunk 1]
```diff
@@ -20212,6 +20212,76 @@ function Remove-NsxIpSet {
...
+
+    .EXAMPLE
+    PS C:\> Get-NsxIPPool TestIPPool | Remove-NsxIPPool
+
+    #>
+ 
+    param (
+
+        [Parameter (Mandatory=$true,ValueFromPipeline=$true,Position=1)]
+            [ValidateNotNullOrEmpty()]
```
Comment (nmbradford) [inline PowerNSX.psm1]: Please validate against the existing Validate-IpPool validation script ie :             [ValidateScript({ Validate-IpPool $_ })]

File: PowerNSX.psm1 [hunk 2]
```diff
@@ -20212,6 +20212,76 @@ function Remove-NsxIpSet {
...
+            [System.Xml.XmlElement]$IPPool,
+        [Parameter (Mandatory=$False)]
+            #Prompt for confirmation.  Specify as -confirm:$false to disable confirmation prompt
+            [switch]$Confirm=$true,
+        [Parameter (Mandatory=$False)]
+            #PowerNSX Connection object.
+            [ValidateNotNullOrEmpty()]
+            [PSCustomObject]$Connection=$defaultNSXConnection
+
```
Comment (nmbradford) [inline PowerNSX.psm1]: Need a switch for 'force' action

File: PowerNSX.psm1 [hunk 3]
```diff
@@ -20212,6 +20212,76 @@ function Remove-NsxIpSet {
...
+    )
+    
+    begin {
+
+    }
+
+    process {
+
+        if ($ippool.SelectSingleNode("descendant::extendedAttributes/extendedAttribute[name=`"isReadOnly`" and value=`"true`"]")) {
```
Comment (nmbradford) [inline PowerNSX.psm1]: Due to differences in the way XML handling must be handled on PowerShell Core, we cant use the native .SelectSingleNode (or SelectNodes) method of the XMLElement.  There is an abstraction function introduced called invoke-xpathquery to allow this to work on either platform.  Need to change this query to utilise invoke-xpathquery.

File: PowerNSX.psm1 [hunk 4]
```diff
@@ -20212,6 +20212,76 @@ function Remove-NsxIpSet {
...
+
+                $choices = New-Object Collections.ObjectModel.Collection[Management.Automation.Host.ChoiceDescription]
+                $choices.Add((New-Object Management.Automation.Host.ChoiceDescription -ArgumentList '&Yes'))
+                $choices.Add((New-Object Management.Automation.Host.ChoiceDescription -ArgumentList '&No'))
+
+                $decision = $Host.UI.PromptForChoice($message, $question, $choices, 1)
+            }
+            else { $decision = 0 } 
+            if ($decision -eq 0) {
+                $URI = "/api/2.0/services/ipam/pools/$($IPPool.objectId)?force=false"
```
Comment (nmbradford) [inline PowerNSX.psm1]: dont hardcode force=false - expose the option to the user via an additional param above.

**Comment by nmbradford** [comment]
actually - i take back the comment 'there are no ip pool tests yet'.  @pandom has submitted a PR adding them.  Ill review shortly and merge if ok.  Ill update this review when I have and you can just add a new test to it.

File: PowerNSX.psm1 (hunk 2 above)
Comment (rhancock8) [inline PowerNSX.psm1]: Hi Nick,  This is where I show my ignorance of git/github <embarassed>
I've tried to understand the instructions on the contribution page, but I use the github desktop for windows GUI so it doesn't really make much sense to me.. sorry.
I have edited the code as per your suggestions (and have committed). Will add the test when you've approved @Pandom tests (to give me an idea of what they should look like).

One quick question... How do I now submit these additional changes back in? .. do I do another PR? or is there a way of updating the existing PR with changes?

Comment (nmbradford) [inline PowerNSX.psm1]: No problem - it wasn't that long ago that I was was trying to comprehend the same thing! Just update the branch in your fork (rhancock8:master) with new commits and they should automatically get added to this PR.

Comment (nmbradford) [inline PowerNSX.psm1]: in fact - had I read any further Id realise thats precisely what youve done!  Ill review first thing tomorrow.

**Comment by nmbradford** [comment]
Not sure why I left this PR open - all changes are merged?  @rhancock8, Im not missing anything am I?

**Pull request merged**